public/services/request-handler.js

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import { HTTP_STATUS_CODES } from '../../common/constants';
+
+let allow = true;
+const source = axios.CancelToken.source();
+
+const disableRequests = () => {
+    allow = false;
+    source.cancel('Requests cancelled');
+    return;
+}
+
+export const initializeInterceptor = (core) => {
+    core.http.intercept({
+        responseError: (httpErrorResponse, controller) => {
+            if (
+                httpErrorResponse.response?.status === 401
+            ) {
+                disableRequests();
+            }
+        },
+    });
+}
+
+export const request = async (options = '') => {
+    if (!allow) {
+        return Promise.reject('Requests are disabled');
+    }
+    if (!options.method | !options.url) {
+        return Promise.reject("Missing parameters")
+    }
+    options = {
+        ...options, cancelToken: source.token, validateStatus: function (status) {
+            return (status >= 200 && status < 300) || status === 401;
+        },
+    }
+    if (allow) {
+        try {
+            const requestData = await axios(options);
+            if (requestData.status === HTTP_STATUS_CODES.UNAUTHORIZED) {
+                if (requestData.data.message === 'Unauthorized' || requestData.data.message === 'Authentication required') {
+                    disableRequests();
+                }
+                throw new Error(requestData.data.message)
```

Comment:
**nitpick**: This modification of the error is breaking the functionality to refresh the page when the session expires.

The valid requests through axios and the returned interface should match with other services or adapt them to the new management.

For the expired session case, the WzRequest and GenericRequest services manage the response and reload the page if the error matches with some conditions. These conditions are being modified with these changes. We should ensure the browser tab is reloaded when the ODFE session expired.

References:
- WzRequest error management related to the ODFE session: https://github.com/wazuh/wazuh-kibana-app/blob/v4.3.9-7.10.2/public/react-services/wz-request.ts#L67
- GenericRequest error management related to the ODFE session: https://github.com/wazuh/wazuh-kibana-app/blob/v4.3.9-7.10.2/public/react-services/generic-request.js#L93
- Reload the browser tab depending on the error message from the request: https://github.com/wazuh/wazuh-kibana-app/blob/v4.3.9-7.10.2/public/utils/odfe-utils.ts#L27-L32

Reply:
Possibly fixed in https://github.com/wazuh/wazuh-kibana-app/pull/4831/commits/030c1e4be75bf725416108c34ff330ed223e9c3e